editor/src/clj/editor/app_view.clj

```diff
@@ -749,6 +751,10 @@
         (swap! initial-output str line "\n")
         (when-let [target-info (engine/parse-launched-target-info @initial-output)]
           (targets/update-launched-target! launched-target target-info on-service-url-found)))
+      (when @engine-just-rebooted
+        (reset! engine-just-rebooted false)
+        (doseq [l-target (targets/all-launched-targets)]
+          (on-service-url-found l-target)))
```

Comment:
Shouldn't the code call `on-service-url-found` only for the `launched-target` that is an argument to this function?

Reply:
Does it work if the old engine is constantly outputting something into logs? Then this branch might be executed before the engine is actually restarted. There is code above (`engine/parse-launched-target-info`) that parses log output, perhaps we should do something like that instead? maybe even without the global state `engine-just-rebooted` atom...

Reply:
I don't like the idea with parsing, because then we have to parse it every single line which is far way from perfect. For now we parse only 5000 messages (which is too many I think).

> Shouldn't the code call on-service-url-found only for the launched-target that is an argument to this function?

no, because it will be changed in `targets/update-launched-target!` 
but I'll recheck logic with multiple instance again (when I have time)